unitTests/ui/DynamicFacet/DynamicFacetTest.ts

```diff
@@ -570,6 +580,55 @@ export function DynamicFacetTest() {
       expect(ResultListUtils.scrollToTop).not.toHaveBeenCalledWith(test.cmp.root);
     });
 
+    describe('Testing the header', () => {
+      beforeEach(() => {
+        test.cmp.selectValue(mockFacetValues[0].value);
+        test.cmp.ensureDom();
+        spyOn(test.cmp.header, 'showLoading').and.callThrough();
+        spyOn(test.cmp.header, 'hideLoading').and.callThrough();
+        spyOn(test.cmp.header, 'toggleClear').and.callThrough();
+        spyOn(test.cmp.header, 'toggleCollapse').and.callThrough();
+      });
+
+      it(`when triggering the header "clear" method
+      should perform the correct actions on the facet`, () => {
+        test.cmp.header.options.clear();
+        expect(test.cmp.reset).toHaveBeenCalledTimes(1);
+        expect(test.cmp.enableFreezeFacetOrderFlag).toHaveBeenCalledTimes(1);
+        expect(test.cmp.triggerNewQuery).toHaveBeenCalledTimes(1);
+        expect(test.cmp.scrollToTop).toHaveBeenCalledTimes(1);
+      });
+
+      it(`when triggering a query
+      should call "showLoading" on the header`, () => {
+        test.cmp.triggerNewQuery();
+        expect(test.cmp.header.showLoading).toHaveBeenCalledTimes(1);
+      });
+
+      it(`when a query is successful
+      should call "hideLoading" on the header`, () => {
+        $$(test.env.root).trigger(QueryEvents.querySuccess, { results: fakeResultsWithFacets() });
+        expect(test.cmp.header.hideLoading).toHaveBeenCalledTimes(1);
+      });
+
+      it(`when triggering the header "clear" method
+      should log an analytics event`, () => {
+        triggerNewQuerySpy.and.callFake((beforeCb: any = () => {}) => beforeCb())
+        test.cmp.header.options.clear();
+        expect(test.cmp.logAnalyticsEvent).toHaveBeenCalledWith(analyticsActionCauseList.dynamicFacetClearAll, test.cmp.basicAnalyticsFacetState);
+      });
+
+      it('should call "toggleClear" when appearance is updated', () => {
```

Comment:
For this and the next test descriptions, would it make more sense to mention the public method name (e.g. `...when calling  reset`, `...when calling collapse`)?